Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -9071,6 +9075,11 @@ void WebPageProxy::showMediaControlsContextMenu(FloatRect&& targetFrame, Vector<
 void WebPageProxy::requestNotificationPermission(const String& originString, CompletionHandler<void(bool allowed)>&& completionHandler)
 {
     auto origin = API::SecurityOrigin::createFromString(originString);
+
+    // Add origin to list of origins that have requested permission to use the Notifications API.
+    WebCore::SecurityOriginData securityOriginData = origin.get().securityOrigin();
+    m_notificationsRequesters.add(securityOriginData);
```

Comment:
We decided not to do this in the permission handler since WebPageProxy::requestPermission() is not called if Notification::requestPermission() is called without a user gesture. So adding an origin to the HashSet as soon as they call requestPermission after a user gesture is correct behavior.